Crypto/ConstructHash/MiyaguchiPreneel.hs

```diff
@@ -0,0 +1,76 @@
+-- |
+-- Module      : Crypto.ConstructHash.MiyaguchiPreneel
+-- License     : BSD-style
+-- Maintainer  : Kei Hibino <ex8k.hibino@gmail.com>
+-- Stability   : experimental
+-- Portability : unknown
+--
+-- provide the hash function construction method from block cipher
+-- <https://en.wikipedia.org/wiki/One-way_compression_function>
+--
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module Crypto.ConstructHash.MiyaguchiPreneel
+       ( mp, mp'
+       , MiyaguchiPreneel
+       , cipherInit'
+       ) where
+
+import           Data.List (foldl')
+
+import           Crypto.Cipher.Types
+import           Crypto.Error (eitherCryptoError)
+import           Crypto.Internal.ByteArray (ByteArrayAccess, ByteArray, Bytes)
+import qualified Crypto.Internal.ByteArray as B
+
+
+newtype MiyaguchiPreneel a = MP Bytes
+    deriving (ByteArrayAccess)
+
+instance Eq (MiyaguchiPreneel a) where
+    MP b1 == MP b2  =  B.constEq b1 b2
+
+
+-- | Compute Miyaguchi-Preneel one way compress using the supplied block cipher.
+mp' :: (ByteArrayAccess bin, BlockCipher cipher)
```

Comment:
I don't think `mp'` is a good name for this, specially as this is the exported function. What about something more self-documenting like `miyaguchiPreneel`, or provided you advise to use the module qualified something like `compute` (my preferred solution)